libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/summary/list/InstructionListAdapter.java

```diff
@@ -13,27 +14,22 @@
 import com.mapbox.api.directions.v5.models.RouteLeg;
 import com.mapbox.services.android.navigation.ui.v5.R;
 import com.mapbox.services.android.navigation.v5.navigation.NavigationConstants;
-import com.mapbox.services.android.navigation.v5.navigation.NavigationUnitType;
 import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress;
 import com.mapbox.services.android.navigation.v5.utils.DistanceUtils;
 
-import java.text.DecimalFormat;
 import java.util.ArrayList;
 import java.util.List;
 
 public class InstructionListAdapter extends RecyclerView.Adapter<InstructionViewHolder> {
-
+  private final Context context;
   private List<LegStep> stepList;
-  private DecimalFormat decimalFormat;
-
   private RouteLeg currentLeg;
   private LegStep currentStep;
-  private LegStep currentUpcomingStep;
-  private int unitType;
+  private DistanceUtils distanceUtils;
 
-  public InstructionListAdapter() {
+  public InstructionListAdapter(Context context) {
     stepList = new ArrayList<>();
-    decimalFormat = new DecimalFormat(NavigationConstants.DECIMAL_FORMAT);
+    this.context = context;
```

Comment:
Rather than store `Context` here as a field, would it be okay to create the instance of `DistanceUtils` here with the `Context` and assign that to a field? `Context` can cause leaks sometimes when storing it like this.